docs/architecture/how/cross-shard.md

```diff
@@ -0,0 +1,301 @@
+            ),
+            header: V3(
+                ShardChunkHeaderV3 {
+                    inner: V2(
+                        ShardChunkHeaderInnerV2 {
+                            prev_block_hash: 82dKeRnE262qeVf31DXaxHvbYEugPUDvjGGiPkjm9Rbp,
+                            prev_state_root: DpsigPFeVJDenQWVueGKyTLVYkQuQjeQ6e7bzNSC7JVN,
+                            outcome_root: H34BZknAfWrPCcppcHSqbXwFvAiD9gknG8Vnrzhcc4w,
+                            encoded_merkle_root: 3NDvQBrcRSAsWVPWkUTTrBomwdwEpHhJ9ofEGGaWsBv9,
+                            encoded_length: 149,
+                            height_created: 1677,
+                            shard_id: 0,
+                            gas_used: 223182562500,
+                            gas_limit: 1000000000000000,
+                            balance_burnt: 22318256250000000000,
+                            outgoing_receipts_root: Co1UNMcKnuhXaHZz8ozMnSfgBKPqyTKLoC2oBtoSeKAy,
+                            tx_root: 11111111111111111111111111111111,
+                            validator_proposals: [],
+                        },
+                    ),
+                    height_included: 0,
+                    signature: ed25519:32hozA7GMqNqJzscEWzYBXsTrJ9RDhW5Ly4sp7FXP1bmxoCsma8Usxry3cjvSuywzMYSD8HvGntVtJh34G2dKJpE,
+                    hash: ChunkHash(
+                        DoF7yoCzyBSNzB8R7anWwx6vrimYqz9ZbEmok4eqHZ3m,
+                    ),
+                },
+            ),
+            transactions: [],
+            receipts: [
+                Receipt {
+                    predecessor_id: AccountId(
+                        "shard0",
+                    ),
+                    receiver_id: AccountId(
+                        "shard1",
+                    ),
+                    receipt_id: 3EtEcg7QSc2CYzuv67i9xyZTyxBD3Dvx6X5yf2QgH83g,
+                    receipt: Action(
+                        ActionReceipt {
+                            signer_id: AccountId(
+                                "shard0",
+                            ),
+                            signer_public_key: ed25519:Ht8EqXGUnY8B8x7YvARE1LRMEpragRinqA6wy5xSyfj5,
+                            gas_price: 103000000,
+                            output_data_receivers: [],
+                            input_data_ids: [],
+                            actions: [
+                                Transfer(
+                                    TransferAction {
+                                        deposit: 500000000000000000000000000,
+                                    },
+                                ),
+                            ],
+                        },
+                    ),
+                },
+            ],
+        },
+    ),
+)
+```
+
+```
+Side comment: notice that receipt itself no longer has a ``signer`` field, but a ``predecessor_id`` one.
+```
+
+Such receipt is magically (we'll explain this magic in a separate article) sent to the destination shard, where it can be executed.
```

Comment:
Personal preference to avoid magic in designs and documentation.

Reply:
ok :-(